### ISSUE TITLE
Thumbnail removes after uploading

### ISSUE THREAD

**Issue by karpo518** [kylegilman/video-embed-thumbnail-generator#430]
Hi. Thanks for plugin!!! I have a problem in version 4.6.22. I had problem with live thumbnail switching(https://wordpress.org/support/topic/err_cache_write_failure/), but i needed to set thumbnail.  I prepeared thumb manually. But i cant update it.

I run "Choose[thumbnail] from Library and oploaded thumb from my PC. I pressed "set as video thumbnail" and got broken link http://joxi.ru/E2pW3VYf7V76Nm

Plugin deleted thumb after uploading. I got php error in log:
`2019/03/23 22:07:46 [error] 14359#14359: *36482580 FastCGI sent in stderr: "PHP message: PHP Warning:  filesize(): stat failed for /wp-content/uploads/2019/03/thumb_tmp/new_poster95.png in /wp-content/plugins/video-embed-thumbnail-generator/video-embed-thumbnail-generator.php on line 6509
PHP message: PHP Warning:  array_key_exists() expects parameter 2 to be array, boolean given in /wp-content/plugins/video-embed-thumbnail-generator/video-embed-thumbnail-generator.php on line 6717
PHP message: PHP Warning:  array_key_exists() expects parameter 2 to be array, boolean given in /wp-content/plugins/video-embed-thumbnail-generator/video-embed-thumbnail-generator.php on line 6717
PHP message: PHP Warning:  array_key_exists() expects parameter 2 to be array, boolean given in /wp-content/plugins/video-embed-thumbnail-generator/video-embed-thumbnail-generator.php on line 3787
PHP message: PHP Warning:  array_key_exists() expects parameter 2 to be array, boolean given in /wp-content/plugins/video-embed-thumbnail-generator/video-embed-thumbnail-generator.php on line 3787
PHP message: PHP Warning:  array_key_exists() expects parameter 2 to be array, boolean given in /wp-content/plugins/video-embed-thumbnail-generator/video-embed-thumbnail-generator.php on line 3787
PHP message: PHP Warning:  array_key_exists() expects parameter 2 to be array, boolean given in /wp-content/plugins/video-embed-thumbnail-generator/video-embed-thumbnail-generator.php on line 3787
PHP message: PHP Warning:  array_key_exists() expects parameter 2 to be array, boolean given in /wp-content/plugins/video-embed-thumbnail-generator/video-embed-thumbnail-generator.php on line 3787" while re`

I fixed it with a temporary patch. I comment lines 6504-6517 in file video-embed-thumbnail-generator.php

It is works for me, but it is not solution. I dont undestand commented code

**Comment by kylegilman** [comment]
I'll get to work on this tomorrow. I can see what the problem is but I can't figure out how to replicate it yet. Is the file very high resolution? Are you generating a lot of thumbnails? 

What you commented out shouldn't cause you too many problems. But if you want to you could always go back to the old version of the plugin. You can download previous version of the plugin at the bottom of this page https://wordpress.org/plugins/video-embed-thumbnail-generator/advanced/

**Comment by karpo518** [comment]
"err_cache_write_failure" - it may be depends with perfomance
 nginx or browser. I realy use big video files and generate a lot of thumbs(99). But it is not this this issue's subject.  Can it be depended?

I wanted fix manual thumb uploading now. I disable all options, which could run background tasks in this plugin. I uploaded one image only. But it was removed at the same moment.

P.S. My code fix above does not works. It fix thumb uploading, but broke video uploading. I returned 4.6.20. It realy works for me.

I do not have enough time for experiments, but if it appears, I will check the latest version of the plugin with minimal server load.
